src/main/java/com/artillexstudios/axenvoy/envoy/Envoys.java

```diff
@@ -18,47 +20,52 @@ public class Envoys {
-        if (CRATE_TYPES_FOLDER.mkdirs()) {
-            FileUtils.copyFromResource("envoys");
-        }
-
-        Collection<File> files = org.apache.commons.io.FileUtils.listFiles(CRATE_TYPES_FOLDER, new String[]{"yaml", "yml"}, true);
-
-        for (File file : files) {
-            Envoy envoy = TYPES.get(file.getName()
-                    .replace(".yml", "")
-                    .replace(".yaml", ""));
-
-            if (envoy == null) {
-                new Envoy(file);
-            } else {
-                envoy.reload();
-            }
-        }
-
-        ArrayList<Envoy> removedTypes = new ArrayList<>();
-        TYPES.entrySet().removeIf((entry) -> {
-            boolean contains = files.contains(entry.getValue().getFile());
-
-            if (!contains) {
-                removedTypes.add(entry.getValue());
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                if (CRATE_TYPES_FOLDER.mkdirs()) {
+                    FileUtils.copyFromResource("envoys");
+                }
+
+                Collection<File> files = org.apache.commons.io.FileUtils.listFiles(CRATE_TYPES_FOLDER, new String[]{"yaml", "yml"}, true);
+
+                for (File file : files) {
+                    Envoy envoy = TYPES.get(file.getName()
+                            .replace(".yml", "")
+                            .replace(".yaml", ""));
+
+                    if (envoy == null) {
+                        new Envoy(file);
+                    } else {
+                        envoy.reload();
+                    }
+                }
+
+                ArrayList<Envoy> removedTypes = new ArrayList<>();
+                TYPES.entrySet().removeIf((entry) -> {
+                    boolean contains = files.contains(entry.getValue().getFile());
+
+                    if (!contains) {
+                        removedTypes.add(entry.getValue());
+                    }
+
+                    return !contains;
+                });
+
+                List<Envoy> envoyTypes = new ArrayList<>(Envoys.getTypes().values());
+                int chestSize = envoyTypes.size();
+
+                for (int i = 0; i < chestSize; i++) {
+                    Envoy envoy = envoyTypes.get(i);
+
+                    if (removedTypes.contains(envoy)) {
+                        envoy.stop();
+                    } else {
+                        envoy.reload();
+                    }
+                }
             }
-
-            return !contains;
-        });
-
-        List<Envoy> envoyTypes = new ArrayList<>(Envoys.getTypes().values());
-        int chestSize = envoyTypes.size();
-
-        for (int i = 0; i < chestSize; i++) {
-            Envoy envoy = envoyTypes.get(i);
-
-            if (removedTypes.contains(envoy)) {
-                envoy.stop();
-            } else {
-                envoy.reload();
-            }
-        }
+        }.runTaskAsynchronously(AxEnvoyPlugin.getInstance());
```

Comment:
This approach won't cut it, as we support Folia. Use the Scheduler.get().runAsync(Runnable) method.